Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -6458,6 +6708,1034 @@ def get_list_asset_group_command(
+        if buffer:
+            yield buffer.decode("utf-8")
+
+        demisto.debug("Finished streaming report.")
+
+    finally:
+        # Crucial: Always ensure the response object is released/closed
+        await response.release()
+
+
+async def stream_and_parse_report(
+    client: "InsightVMClient",
+    report_id: str,
+    instance_id: str,
+    event_integration_context: dict,
+    event_type: str,
+    batch_size: int = 1000,
+):
+    # --- Checkpoint Setup ---
+    # Retrieve checkpoints:
+    last_sent_line_raw = event_integration_context.get("last_sent_line", 0)
+    total_records_ingested = event_integration_context.get("total_records_ingested", 0)
+    header: Optional[List[str]] = None
+    start_line_for_batch_raw = last_sent_line_raw + 1
+    pending_tasks: set[asyncio.Task] = set()
+
+    demisto.debug(f"--- Starting Data Stream (Resuming from line: {start_line_for_batch_raw}) ---")
+    # --- Lookahead Stream Setup ---
+    try:
+        stream_iterator = stream_report(client, report_id, instance_id).__aiter__()
+    except Exception as e:
+        raise DemistoException(f"Failed to initialize stream: {e}")
+
+    line_to_process: Optional[str] = None
+
+    demisto.debug(f"--- Starting Data Stream for {event_type} report instance {instance_id} ---")
+
+    try:
+        # INITIAL LINE READ: Get the header of the report
+        try:
+            # Fetch the very first line (Line 1)
+            header_line = await anext(stream_iterator)
+
+            # Parse the header immediately
+            header = next(csv.reader(io.StringIO(header_line)))
+            demisto.debug(f"CSV Header parsed: {header}")
+
+        except StopAsyncIteration:
+            demisto.debug("Report was empty (no header found).")
+            return
+        except Exception as e:
+            raise Exception(f"Failed to parse CSV header on line 1: {e}")
+
+        try:
+            # Try to fetch Line 2 (the first data line)
+            line_to_process = await anext(stream_iterator)
+        except StopAsyncIteration:
+            # Report has a header but no data
+            line_to_process = None
+
+        # --- EXECUTION PHASE ---
+        if line_to_process:
+            await process_data_stream(
+                stream_iterator=stream_iterator,
+                line_to_process=line_to_process,
+                header=header,
+                last_sent_line_raw=last_sent_line_raw,
+                total_records_ingested=total_records_ingested,
+                batch_size=batch_size,
+                event_type=event_type,
+                pending_tasks=pending_tasks,
+                event_integration_context=event_integration_context,
+                instance_id=instance_id,
+            )
+        else:
+            demisto.debug("No data rows to process.")
+
+    except Exception as e:
+        raise DemistoException(f"\nFATAL ERROR during streaming or sending events: {e}")
+
+
+async def process_data_stream(
```

Comment:
Wayyy to long function to follow
Try to break it into smaller helper functions like:
_parse_csv_row(...)
_should_skip_line(...)
_submit_batch(...)